### ISSUE TITLE
Configuring vsgvr inside vsg-dev/vsgFramework fails

### ISSUE THREAD

**Issue by rhabacker** [geefr/vsgvr#12]
On openSUSE Leap I did run the following commands:

```
mkdir ~/src && cd ~/src
git clone https://github.com/vsg-dev/vsgFramework.git
cd vsgFramework
cat << EOF >> CMakeLists.txt
##############################################################################################################
#
# vsgVr
#
OPTION(BUILD_vsgVR "Build vsgVR" ON)
if (${BUILD_vsgVR})
    FetchContent_Declare(vsgvr
        GIT_REPOSITORY "https://github.com/geefr/vsgvr.git"
        GIT_TAG "main"
        GIT_PROGRESS TRUE
    )
    FetchContent_MakeAvailable(vsgvr)
endif()
EOF
```
and got the following failure when running  `cmake -B ../vsgFramework-build` 
```
...
CMake Error at /home/user/src/vsgFramework-build/components/vsgvr-src/CMakeLists.txt:17 (find_package):
  find_package for module vsg called with REQUIRED, but
  CMAKE_DISABLE_FIND_PACKAGE_vsg is enabled.  A REQUIRED package cannot be
  disabled.
```

**Comment by geefr** [comment]
Why is find_vsg being disabled?
That's a required package for vsgvr to configure/build..

**Comment by rhabacker** [comment]
I wrote that in the commit at https://github.com/geefr/vsgvr/pull/13

> vsgFramework adds vsgvr as an additional subproject where the vsg project is already available as a cmake target and find_package is not usable.

**Comment by geefr** [comment]
Seems like a strange way to correct the issue. Adding external modules should not require modifications to the included project.

How does vsgExamples do this?
Why can't vsgFramework pass in the vsg location when configuring vsgvr? - vsg must be compiled before vsgvr, so should have produced a vsgconfig.cmake file as part of that.

**Comment by rhabacker** [comment]
They removed the REQUIRED statement, see https://github.com/vsg-dev/vsgExamples/commit/aede11f0a8db8e23073ecd59e5d8afd38b8a4e2b

**Comment by rhabacker** [comment]
> CMAKE_DISABLE_FIND_PACKAGE_vsg

This cmake variable was introduced with https://github.com/vsg-dev/VulkanSceneGraph/commit/0abaf64108e86fc3db28671f6396ae751305152b

**Comment by rhabacker** [comment]
> Why can't vsgFramework pass in the vsg location when configuring vsgvr? - vsg must be compiled before vsgvr, so should have produced a vsgconfig.cmake file as part of that.

The vsgFramework project has recently been changed from `ExternalProject` to `FetchContent`, where each project is added as a submodule and a large cmake project is created. vsgvr is built as part of the `cmake --build` process. 

find_package needs additional files besides vsgConfig.cmake, which are only created during an installation and thus are not available during the build step.
```
user@host:~/src/vsgFramework-build> make install DESTDIR=../vsgFramework-install 
user@host:~/src/vsgFramework-build> ls ../vsgFramework-install/usr/local/lib64/cmake/vsg -l
total 20
-rw-r--r-- 1 user users 767 12 Oct 16:29 FindVulkan.cmake
-rw-r--r-- 1 user users 490 12 Oct 16:29 vsgConfig.cmake
-rw-r--r-- 1 user users 1722 12 Oct 16:29 vsgConfigVersion.cmake
-rw-r--r-- 1 user users 3626 12 Oct 16:30 vsgTargets.cmake
-rw-r--r-- 1 user users 803 13 Oct 12:03 vsgTargets-release.cmake
```

**Comment by geefr** [comment]
FetchContent should be installing vsg to a directory then, as one would if they did a FetchContent(fmt) or other open source project.

Projects included this way really should not be modified because of the parent project's quirks - removing the REQUIRED in vsgvr for instance means it will then fail whenever vsg isn't available from say a package manager.

Apologies, I have very minimal time at the moment, and am in the middle of several crunch times :) I'd definitely say the issue is in vsgFramework, afraid I don't have a 'correct' example to share at the moment.

**Comment by robertosfield** [comment]
I spent days trying to avoid the problems with FetchContent, what is checked in is the best workaround I could come up with, it looks like a CMake problem.

I'm not personally going to waste any more time on the problem within the VSG project, if you want to invest your own time on the issue please feel free.  Personally I'd recommend just getting just removing the REQUIRED option as a workaround as there are so many more important real-world problems you can spend your time on.

**Comment by geefr** [comment]
@robertosfield Fair enough, folks are free to branch as needed for now, I'll take a look/merge when I can.

(I expect other vsg changes have broken vsgvr, since my integration was still fragile. No rush on this stuff from my perspective..)

**Comment by geefr** [comment]
Okay, thanks for the fix.

Note that vsgvr doesn't have cmake install targets yet - I've just been running from the build directory for now

**Comment by rhabacker** [comment]
Since I often use packages, I made that possible on https://build.opensuse.org/package/show/home:rhabacker:branches:games/vsgvr that with a temporary patch.
